internal/k8s/client.go

```diff
@@ -540,11 +542,23 @@ var MetadataAnnotationsTooLongRe = regexp.MustCompile(`metadata.annotations: Too
 // However, annotations have a max size of 256k. Large objects such as configmaps can exceed 256k, which makes
 // apply unusable, so we need to fall back to delete/create
 // https://github.com/kubernetes/kubectl/issues/712
-func maybeAnnotationsTooLong(stderr string) (string, bool) {
+//
+// We've also seen this reported differently, with a 413 HTTP error.
+// https://github.com/tilt-dev/tilt/issues/5279
+func maybeTooLargeError(err error) (string, bool) {
+	statusErr, isStatusErr := err.(*apiErrors.StatusError)
+	if isStatusErr && statusErr.ErrStatus.Code == http.StatusRequestEntityTooLarge {
+		return err.Error(), true
+	}
+
+	stderr := err.Error()
 	for _, line := range strings.Split(stderr, "\n") {
 		if MetadataAnnotationsTooLongRe.MatchString(line) {
 			return line, true
 		}
+		if strings.Contains(line, "the server responded with the status code 413") {
```

Comment:
sigh

Reply:
ya, unfortunately, a lot of the Go ecosystem doesn't do a great job preserving error types, so it's generally always good to check error messages